src/H5VLpassthru.c

```diff
@@ -2209,31 +2181,31 @@ H5VL_pass_through_link_create(H5VL_link_create_type_t create_type, void *obj,
 
     /* Fix up the link target object for hard link creation */
     if (H5VL_LINK_CREATE_HARD == create_type) {
-        void *             cur_obj;
-        H5VL_loc_params_t *cur_params;
+        void *cur_obj;
 
         /* Retrieve the object & loc params for the link target */
-        cur_obj    = va_arg(arguments, void *);
-        cur_params = va_arg(arguments, H5VL_loc_params_t *);
+        cur_obj = create_args->hard.vol_obj_data;
 
         /* If it's a non-NULL pointer, find the 'under object' and re-set the property */
         if (cur_obj) {
+            /* Copy the current link creation arguments */
+            new_create_args = *create_args;
```

Comment:
Since the VOL object needs to be modified here, we just take a copy of the const args and pass the copy down. The var arg logic is mostly just substituted with logic to determine which of the arguments we need to pass down (the original or the copy), but it removes the need for the _reissue function and makes the reissue operation look consistent with other callbacks.